pyproject.toml

```diff
@@ -27,8 +27,34 @@ line-length = 130
 target-version = "py39"
 
 [tool.ruff.lint]
+# TODO: Use extend-select instead to get base E and F rules that don't conflict with the formatter
 select = [
+    "FA", # flake8-future-annotations
     "I", # isort
+    "PYI", # flake8-pyi
+]
+ignore = [
+    ###
+    # Rules we don't want or don't agree with
+    ###
+    # Typeshed doesn't want complex or non-literal defaults for maintenance and testing reasons.
+    # This doesn't affect us, let's have more complete stubs.
+    "PYI011",
+    "PYI014",
+
+    # TODO: Handle in its own PR
+    "PYI021", # https://github.com/microsoft/python-type-stubs/pull/343
+
+    # TODO: Investigate and fix or configure
+    "PYI001",
+    "PYI002",
+    "PYI017",
+    "PYI019", # Request for more autofixes: https://github.com/astral-sh/ruff/issues/14183
+    "PYI024",
+    "PYI048",
+    "PYI051", # Request for autofix: https://github.com/astral-sh/ruff/issues/14185
+    "PYI052",
+    "PYI061", # Request for more autofixes: https://github.com/astral-sh/ruff/issues/14537
```

Comment:
Updated references for Ruff 0.8.0:
- https://github.com/astral-sh/ruff/issues/14183
- https://github.com/astral-sh/ruff/issues/14185
- https://github.com/astral-sh/ruff/issues/14537

PYI021 is also done separately in https://github.com/microsoft/python-type-stubs/pull/343